code/modules/projectiles/projectile/magic.dm

```diff
@@ -434,43 +434,35 @@
 /obj/structure/closet/decay
 	breakout_time = 600
 	icon_welded = null
-	var/magic_icon = "cursed"
+	icon_state = "cursed"
 	var/weakened_icon = "decursed"
 	var/auto_destroy = TRUE
 
 /obj/structure/closet/decay/Initialize()
 	. = ..()
 	if(auto_destroy)
 		addtimer(CALLBACK(src, .proc/bust_open), 5 MINUTES)
-	addtimer(CALLBACK(src, .proc/magicly_lock), 5)
-
-/obj/structure/closet/decay/proc/magicly_lock()
-	if(!welded)
-		return
-	icon_state = magic_icon
-	update_icon()
 
 /obj/structure/closet/decay/after_weld(weld_state)
 	if(weld_state)
 		unmagify()
 
+///Fade away into nothing
 /obj/structure/closet/decay/proc/decay()
 	animate(src, alpha = 0, time = 30)
 	addtimer(CALLBACK(GLOBAL_PROC, .proc/qdel, src), 30)
 
 /obj/structure/closet/decay/open(mob/living/user, force = FALSE)
 	. = ..()
 	if(.)
-		if(icon_state == magic_icon) //check if we used the magic icon at all before giving it the lesser magic icon
-			unmagify()
-		else
-			addtimer(CALLBACK(src, .proc/decay), 15 SECONDS)
+		unmagify()
 
+///Give it the lesses magic icon and tell ot ti delete itself in 15 seconds
```

Comment:
```suggestion
///Give it the lesses magic icon and tell it to delete itself
```

We don't need to document times, it just leads to more work when you want to change it.